### ISSUE TITLE
Incorrect imports suggestion for code inside `tokio::main`

### ISSUE THREAD

**Issue by Cerber-Ursi** [rust-lang/rust#86241]
Given the following [code](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=e9aef2b2cf975ea3fe01841ec7304da2):

```rust
#[tokio::main]
async fn main() {
    let _ = Duration::from_millis(97);
}
```

The current output is:

```
error[E0433]: failed to resolve: use of undeclared type `Duration`
 --> src/main.rs:3:13
  |
3 |     let _ = Duration::from_millis(97);
  |             ^^^^^^^^ not found in this scope
  |
help: consider importing one of these items
  |
2 | async use chrono::Duration;
  |       ^^^^^^^^^^^^^^^^^^^^^
2 | async use core::time::Duration;
  |       ^^^^^^^^^^^^^^^^^^^^^^^^^
2 | async use humantime::Duration;
  |       ^^^^^^^^^^^^^^^^^^^^^^^^
2 | async use instant::Duration;
  |       ^^^^^^^^^^^^^^^^^^^^^^
    and 5 other candidates
```

Note the `async use` bit, which is, obviously, invalid syntax.

**Comment by ehuss** [comment]
Thanks for the report!

Even fixing the `async` part won't be sufficient because the `#[tokio::main]` attribute is in the way, too.  There is more discussion at https://github.com/rust-lang/rust/pull/85312#issuecomment-841704641 and https://github.com/rust-lang/rust/pull/85427 as to why that doesn't work.

**Comment by pnkfelix** [comment]
(This is probably fixed by #90941.)

**Comment by Cerber-Ursi** [comment]
Just checked in playground - seems to be fixed for now, probably should be closed?

**Comment by ehuss** [comment]
Yea, I'm going to close as resolved by #94584.
